ui/components/NodeInputSubmit.tsx

```diff
@@ -43,9 +50,50 @@ export const NodeInputSubmit: FC<NodeInputProps> = ({
     (node.meta.label as unknown as { hasTotpLink: boolean })?.hasTotpLink ??
     false;
   const showBackLink = node.meta?.label?.text === "Reset password";
+  const showBackupCodesIntro =
+    (node.meta.label?.context as unknown as { showBackupCodesIntro: boolean })
+      ?.showBackupCodesIntro ?? false;
+  const hasSavedCodeCheckbox =
+    (node.meta.label?.context as unknown as { hasSavedCodeCheckbox: boolean })
+      ?.hasSavedCodeCheckbox ?? false;
+  const getAppearance = () => {
```

Comment:
+1

Reply:
Suggestion: this might as well be a "regular" function outside of the react component, so it's easier to read and less crowded over here.
If for any reason we prefer to keep it here, than let's use `useMemo` to "store" the reuse the result in case dependencies don't change

Reply:
I think this should stay as is, but we should factor out the specific logic to a higher order. So only have generic logic in here. I'll refactor a bit and we can continue the discussion on this with a later update of the code.